url026791d51b_dhleong_SeriesTracker_v6_tgz-pJ8-stv6_SeriesTrackerJ8/src/stv6/Profile.java

```diff
@@ -0,0 +1,801 @@
+		}
+	}
+	
+	@Override
+    public void reload() {
+		
+		// make sure we only try to reload once
+		synchronized(reloading) {
+			if (!reloading) {
+				reloading = true;
+				new Thread(this, "Profile-Reload").start();
+			}
+		}
+	}
+
+	/**
+	 * Threaded reloading fun
+	 */
+	@Override
+	public void run() {
+		STHandlerManager.getInstance().setStaticMessage(true);
+		StaticMessageHandler.getInstance().setBody();
+		System.out.print("Reloading... ");
+		if (!profileFromCommandLine && !isSelected() && profileFile == null) {
+			System.out.println("Failed (No profile chosen; will prompt)");
+			System.out.println("Server listening on port: " + getPort());
+			synchronized(reloading) {
+				reloading = false;
+			}
+			return;
+		}
+		series.clear();
+		db.reload();
+		
+		globalUser = null; // clear it before anything
+		if (userName != null) {
+			globalUser = db.getUser(userName);
+		} else {
+			// no user specified; check for a single user environment
+			possibleUsers.clear();
+			db.getAllUsers(possibleUsers);
+			if (possibleUsers.size() == 1)
+				globalUser = possibleUsers.get(0);
+			else if (possibleUsers.size() == 0) 
+				globalUser = db.getUser(STClient.DEFAULT_USERNAME);
+		}
+		
+		// determine profile id
+		if (profileName != null) {
+			id = db.getProfileId(profileName);
+		} else if (profileFile != null) {
+			id = db.getProfileId(profileFile);
+		}
+		if (id < 0) {
+			System.err.println("Couldn't determine profile ID");
+			System.exit(1);
+		}
+
+		if (isSyncing()) {
+			if (globalUser != null && globalUser.getName().equals(STClient.DEFAULT_USERNAME)) {				
+				StaticMessageHandler.getInstance().setBody(
+						"To use synchronization, you must create a username. " +
+						"Either specify a user in the profile using \"-u\" or " +
+						"remove synchronization, restart SeriesTracker, and create " +
+						"a user in the Options menu.");
+				StaticMessageHandler.getInstance().setRefresh(null);
+				System.out.println("[No user; could not sync] Failed.");
+				return;
+			}
+			System.out.print("[Synchronizing... ");
+			StaticMessageHandler.getInstance().setBody("Synchronizing data with remote server");
+			try {
+				if (!Synchronizer.getInstance().synchronize(syncSettings))
+					return;
+			} catch (UnknownHostException e) {
+				System.out.print("Failure (no internet/invalid server) ");
+			} catch (IOException e) {
+				e.printStackTrace();
+				System.out.print("Failure (???)");
+			}
+			
+			// reload users list, just in case...
+			// TODO: Have synchronize() return something more meaningful
+			//	so we don't have to do this every time
+			possibleUsers.clear();
+			db.getAllUsers(possibleUsers);
+			
+			System.out.print("Done.] ");
+			StaticMessageHandler.getInstance().setBody();
+		}
+
+		db.fillSeries(series, id);
+		
+		epmgr.reload();
+		epmgr.getAvailableSeries(series);	
+		System.out.println("Done.");
+		STHandlerManager.getInstance().setStaticMessage(false);
+		
+		synchronized(reloading) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Object reloading synchronized on line 581 should not be re-assigned in the same synchronized block on line 582. After this re-assignment, you lose the reference to the synchronized object.